modules/standard/Map.chpl

```diff
@@ -254,6 +257,70 @@ module Map {
       return result;
     }
 
+    pragma "no doc"
+    proc const this(k: keyType)
+    where isNonNilableClass(valType) {
+      compilerError("Cannot access nilable class directly. Use an",
+                    " appropriate accessor method instead.");
+    }
+
+    /* Get a borrowed reference to the element at position `k`.
+     */
+    proc getBorrowed(k: keyType) {
```

Comment:
What about if the value type is not a class? Does it make sense to allow a call to `getBorrowed`? Can we restrict this with something like `where isClass(valType)`?

Reply:
Sure, that where clause sounds reasonable to me.